evap/staff/views.py

```diff
@@ -1051,6 +1051,39 @@ def course_copy(request, course_id):
     )
 
 
+@require_POST
+@manager_required
+@transaction.atomic
+def create_exam_evaluation(request):
+    evaluation = get_object_from_dict_pk_entry_or_logged_40x(Evaluation, request.POST, "evaluation_id")
+    if evaluation.is_single_result:
+        raise SuspiciousOperation("Creating an exam evaluation for a single result evaluation is not allowed")
+
+    if evaluation.has_exam:
+        raise SuspiciousOperation("An exam evaluation already exists for this course")
+    try:
+        exam_datetime = request.POST.get("exam_date")
+        exam_datetime = datetime.combine(datetime.strptime(exam_datetime, "%Y-%m-%d"), datetime.min.time())
+    except ValueError:
+        return HttpResponseBadRequest("Exam date missing or invalid.")
+
+    evaluation_end_date = exam_datetime - timedelta(days=1)
+    if evaluation.vote_start_datetime > evaluation_end_date:
+        messages.error(
+            request, _("The exam date is before the start date of the main evaluation. No exam evaluation was created.")
+        )
+        return HttpResponse()
```

Comment:
I don't think the message flow makes sense here (as is reflected in the test for this): The post request will be answered with an empty HTTP 200 OK -- how would a user get that error? To my understanding, they would only see that once they go to a new page, right?

If this error is possible to be triggered by a user, they should get a meaningful error immediately.

(A post-only REST handler setting an error in the messages framework generall seems not really useful to me)

Reply:
@niklasmohrin we settled to check the date when the user submits the modal, similar to the semester deletion modal (See the new commit). One problem we currently face is the error message disappearing fast as the modal closes on click. How would this feature work best with the confirmation modal?

Reply:
I opened #2253 to cover this case. There is also another approach described in the comments there, however I am not sure which of the two solutions should be preferred. If you want to move forward here, you can copy the solution as in the semester deletion modal